kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -462,13 +409,31 @@ void FindIntersectedGeometricalObjectsProcess<TEntity>::FindIntersectedSkinObjec
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<>
+PointerVectorSet<Element, IndexedObject>::ContainerType& FindIntersectedGeometricalObjectsProcess<Element>::GetIntersectingEntities()
+{
+    return mrModelPartIntersected.ElementsArray();
```

Comment:
Are you sure of this? This is supposed to be the skin (intersecting model part). In here you are taking the elements, or conditions in the method below, from the volume part (intersected model part). If I'm not remembering wrong, the volume part must always be defined in terms of elements.

Reply:
I just changed the name according to you suggestion :P

Reply:
Indeed the method is named `GetIntersectingElements` and you are working with the `mrModelPartIntersected`.

Reply:
> I just changed the name according to you suggestion :P

Well, it is not a suggestion. I guess the distance algorithms wouldn't work with the current implementation.

Reply:
This has to return the intersecting entities, which are the ones that can be defined as elements (as they are now) or as conditions (what we suggested to implement). The intersected model part will never be defined as a condition.

Reply:
Well I just mistake the names..., changing

Reply:
> Well I just mistake the names..., changing

Were the tests working? If they were we need to further improve them...

Reply:
Yes, I only changed the names, nothing else. Just the names are wrong

Reply:
Wouldn't be better to add a `GetEntities` method to the ModelPart? There are other issues with the same request and we can finish all these kind of implementations much easier and cleaner!

Reply:
Done